packages/flutter/lib/src/widgets/scrollable.dart

```diff
@@ -780,7 +780,12 @@ class ScrollableState extends State<Scrollable> with TickerProviderStateMixin, R
         GestureBinding.instance.pointerSignalResolver.register(event, _handlePointerScroll);
       }
     } else if (event is PointerScrollInertiaCancelEvent) {
-      position.jumpTo(position.pixels);
+      if (position is ScrollActivityDelegate) {
```

Comment:
Mainly just curious, in what case would this not be a ScrollActivityDelegate?

Reply:
Nowhere in flutter framework, but I guess they aren't guaranteed to be combined together. I don't know what the impact would be to make `ScrollPosition` implement `ScrollActivityDelegate`. Is there any way of seeing what that would break with whatever googler information you have available to you?

Reply:
Yeah, let me check and see if there are any cases where someone has implemented ScrollPosition and not included ScrollActivityDelegate..

Reply:
Actually, I started digging a bit, and I have a couple of ideas...

First, ScrollPosition has the beginActivity method, so you could actually just call for that here:

```dart
position.beginActivity(IdleScrollActivity(...));
```

Although in that case, it is required to provide a delegate, but...

ScrollActivity (and its ScrollActivityDelegate) manage dragging and flinging, so could it stand to reason that a scroll position that ends up here must be of a ScrollActivityDelegate?

If so, you could assert that...

```dart
assert(position is ScrollActivityDelegate);
position.beginActivity(IdleScrollActivity(position as ScrollActivityDelegate));
```

However, we don't have anything like this in Scrollable as an existing pattern. What we normally do is call on methods like pointerScroll that ScrollPositionWithSingleContext and _NestedScrollPosition implement in their own unique ways.
Ideally I think we would have this handled the same way. A method on ScrollPosition (like maybe `cancelActivity`?) would allow the same convention.
There are a fair number of customers that implement ScrollPosition so it would be breaking to add to ScrollPosition. Many of the implementing cases I can see though are just test mocks. I've fired off a global test run on adding something to ScrollPosition so I'll repot back on how breaking it is.

If `assert(position is ScrollActivityDelegate);` is logically fair, that might be the way to go if adding to ScrollPosition is incredibly breaking. What are your thoughts?